### DISCUSSION TITLE
Can't execute program without mount REU image

### DISCUSSION THREAD

**Discussion by odflor** [frntc/RAD#30] [General]
Hi!:

For some reason, I can't execute a program in the RAD_PRG folder without mounting a ramdon image on the REU folder.

Attached a vídeo to show.

https://drive.google.com/file/d/1OuuqYN9n6VLZYg_Pu3MJxiUgnreCUIzs/view?usp=sharing

I try to execute the tester programs, even the NEO_RAM memory disk utility and I can't because the RAD mount a random REU image.

I have a c128 NTSC "slim" computer.

Best Regards

**Comment by frntc** [comment]
In the video you unmount the REU image and when you press RETURN it mounts it again? I tried, but I'm not able to reproduce this behavior. 

Does the NeoRAM-tester work when you select the GeoRAM as memory expansion?

Btw. I noticed that the timings (pound-sign) need some adjustment as there are some glitches on the screen.

**Reply by odflor** [reply]
1) Yes, when I press RETURN, a random image mount again. I don't know is important, but the programs are in sub-folders.
2) No, no program executes after that.
3) That's right!, I need to know the timings for my c128 NTSC. I try random values, but nothing works. Mainly because I don't know what is each value for.

Best Regards

**Reply by odflor** [reply]
I will try this weekend to execute the tester programs from my sd2iec instead of the RAD itself.

Best Regards

**Reply by frntc** [reply]
I was pointed to another possible root of your problem: do you have many thousands of PRGs/REU/GeoRAMs on your SD-card? The code is bad here and more than 16k files lead to problems...

Regarding timings: yes, hard to interpret -- try adjusting the last three, "Enable RW+ADDR" / "Enable DATA" should be some distance apart but can take very different values (for some machines quite a range works, for others not). The last parameter is a finetuning parameter, iirc reasonable values are approx. 40 to 70.

**Comment by odflor** [comment]
Mr. Frntc:

As You say, my SD have thousands of files inside (reu images, prg games, etc) in severals sub directories .

I make two things and it finally works (but still a little buggy):

1) I remove all files that are not part of the original image downloaded from Github and only remain the geo/reu utilities and one game and one reu image. I don't use any sub directory at all. Besides I remove the "System Volume Information" folder that Windows puts on the SD without asking some sh....

2) I see that in the file /RAD/rad.cfg are some special configuration when gliches appears and activate them after the label
# ####################################################
# if you're using a CARTRIDGE PORT EXPANSION and 
# experience glitches/problems: use the timings below
# ####################################################

Then I turn on all and finally the main menú appears without gliches.

I test to load the game and work.

Later I press de menú button and not work. I press the reset button (of the RAD) and main menú don't start. I press the reset of the c128 and the same occurs. I turn off/on the c128 and still is freeze. Al last turn all off, and on and works again.

I configure the RAD to emulate a REU and execute the reu-test program and the attached screen apperars to me.

![IMG_20231011_081057](https://github.com/frntc/RAD/assets/25032803/9957e137-6e02-4ea7-80d9-0dc4b330da30)

Later I configure the RAD to emulate a GEORAM of 512, start the program to do a RAM disk (neo ram disk) make a basic program, save and verify to the device # 2 and works. On the main menu I save the georam image and later restart all the devices + c128. And when all is on, I can retrieve the basic program from the georam image very well. It works!

In this point, there are things that work and other not. I will continue testing all the features one by one.

Best Regards
